hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java

```diff
@@ -1786,6 +1787,51 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
             () -> store.deleteSnapshot(nullstr, bucket, snap1));
   }
 
+  @Test
+  public void testSnapshotQuotaHandling() throws Exception {
+    String volume = "vol-" + counter.incrementAndGet();
+    String bucket = "buck-" + counter.incrementAndGet();
+    store.createVolume(volume);
+    OzoneVolume volume1 = store.getVolume(volume);
+    volume1.createBucket(bucket);
+    OzoneBucket bucket1 = volume1.getBucket(bucket);
+    bucket1.setQuota(OzoneQuota.parseQuota("102400000", "500"));
+    volume1.setQuota(OzoneQuota.parseQuota("204800000", "1000"));
+    // Create Key1 and take snapshot
+    String key1 = "key-1-";
+    createFileKeyWithPrefix(bucket1, key1);
+
+    long volNameQuotaBefore = volume1.getQuotaInNamespace();
+    long volSpaceQuotaBefore = volume1.getQuotaInBytes();
+    long buckNameQuotaBefore = bucket1.getQuotaInNamespace();
+    long buckSpaceQuotaBefore = bucket1.getQuotaInBytes();
+
+    String snap1 = "snap" + counter.incrementAndGet();
+    createSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaAfter = volume1.getQuotaInNamespace();
+    long volSpaceQuotaAfter = volume1.getQuotaInBytes();
+    long buckNameQuotaAfter = bucket1.getQuotaInNamespace();
+    long buckSpaceQuotaAfter = bucket1.getQuotaInBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaAfter);
+    assertEquals(volSpaceQuotaBefore, volSpaceQuotaAfter);
+    assertEquals(buckNameQuotaBefore, buckNameQuotaAfter);
+    assertEquals(buckSpaceQuotaBefore, buckSpaceQuotaAfter);
+
+    store.deleteSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaFinal = volume1.getQuotaInNamespace();
+    long volSpaceQuotaFinal = volume1.getQuotaInBytes();
+    long buckNameQuotaFinal = bucket1.getQuotaInNamespace();
+    long buckSpaceQuotaFinal = bucket1.getQuotaInBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaFinal);
```

Comment:
The Quota set on the bucket/volume doesn't change on taking /deleting a snapshot, we can do an assert for bucket.getUsedBytes() before and after these operations.

Reply:
Added checks for getUsedBytes and getUsedNamespace for volume and bucket.